Source/Core/Core/MemWatcher.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright 2015 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <atomic>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <sys/socket.h>
+#include <sys/un.h>
+
+#include "Common/FileUtil.h"
+#include "Common/Thread.h"
+#include "Core/MemWatcher.h"
+#include "Core/HW/Memmap.h"
+
+namespace MemoryWatcher
+{
+
+// We don't want to kill the cpu, so sleep for this long after polling.
+static const int SLEEP_DURATION = 10; // ms
+
+static std::thread s_watcher_thread;
+static std::atomic_bool s_running;
+
+static int s_fd;
+static sockaddr_un s_addr;
+
+// Address -> last value.
+static std::map<u32, u32> s_values;
+
+static bool LoadAddresses()
+{
+	std::string locations_path = File::GetUserPath(F_MWCONFIG_IDX);
+	if (!File::Exists(locations_path))
+		return false;
+
+	std::ifstream locations(locations_path);
+	bool okay = false;
+	u32 data;
+	locations >> std::hex >> data;
+	while (locations.good() && !locations.eof())
+	{
+		okay = true;
+		s_values[data] = 0;
+		locations >> std::hex >> data;
+	}
+	return okay;
+}
+
+static bool LoadSocket()
+{
+	std::string sock_path = File::GetUserPath(F_MWSOCK_IDX);
+
+	memset(&s_addr, 0, sizeof(s_addr));
+	s_addr.sun_family = AF_UNIX;
+	strncpy(s_addr.sun_path, sock_path.c_str(), sizeof(s_addr.sun_path) - 1);
+
+	s_fd = socket(AF_UNIX, SOCK_DGRAM, 0);
+	return s_fd >= 0;
+}
+
+static void WatcherThread()
+{
+	s_running = true;
+	while (s_running)
+	{
+		for (auto iter = s_values.begin(); iter != s_values.end(); iter++)
```

Comment:
This can be:

```cpp
for (auto& entry : s_values)
{
    u32 address = entry.first;
    u32& current_value = entry.second;

    ... rest here
}
```